certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -278,29 +317,68 @@ func csvStatusSucceeded(ctx context.Context, engine cli.OpenshiftEngine, name, n
 		log.Debug("CSV is created successfully: ", name)
 		return name, true, nil
 	}
-
 	return "", false, nil
 }
 
-func (p *DeployableByOlmCheck) isCSVReady(ctx context.Context, installedCSV string, operatorData OperatorData) (bool, error) {
+func (p *DeployableByOlmCheck) isCSVReady(ctx context.Context, operatorData OperatorData) (bool, error) {
 
-	log.Trace(fmt.Sprintf("Looking for csv %s in namespace %s", installedCSV, operatorData.InstallNamespace))
+	var CsvNamespaces []string
+	if len(operatorData.CsvNamespaces) == 0 {
+		CsvNamespaces = []string{operatorData.TargetNamespace, "default", openshiftMarketplaceNamespace}
+	} else {
+		CsvNamespaces = []string{operatorData.CsvNamespaces[0]}
+	}
+	log.Trace(fmt.Sprintf("Looking for csv %s in namespace(s) %s", operatorData.InstalledCsv, CsvNamespaces))
 
-	csvReadyDone := make(chan string, 1)
-	defer close(csvReadyDone)
+	doneChannel := make(chan string)
+	errChannel := make(chan error)
 
-	contextTimeout := make(chan error, 1)
-	defer close(contextTimeout)
+	var wg sync.WaitGroup
+	go func() {
+		wg.Wait()
```

Comment:
Does this wait group ever get finished since it's in a different thread than the thread that is adding to the wait group? I'd expect to see this `wg.Wait()` at line 347. This might work, but I've never seen it this way, so I am unsure.

Reply:
yes it does. For the check, we need to read the messages from channels and the goroutine should not be blocked when writing to channels. I found this article useful to explain why wg.wait() has to be in another goroutine when you we want to read messages from multiple goroutine over a shared channel: https://dev.to/sophiedebenedetto/synchronizing-go-routines-with-channels-and-waitgroups-3ke2

Reply:
Awesome! Never seen it coded this way so I was unsure. Thanks for the link!

Reply:
I was confusing `wg.Wait()` and `wg.Done()` I see the `wg.Done()` as a defer in the `watch()` function.